pkg/platform/common/cloudletssh/sshkey.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024 EdgeXR, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cloudletssh provides for a shareable, on-demand provider
+// of signed ssh keys.
+package cloudletssh
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/edgexr/edge-cloud-platform/pkg/log"
+	ssh "github.com/edgexr/golang-ssh"
+	xssh "golang.org/x/crypto/ssh"
+)
+
+// SSHKey provides on-demand signed ssh certificates for authenticating
+// with VMs created by the platform.
+type SSHKey struct {
+	publicKey         string
+	signedPublicKey   string
+	privateKey        string
+	signer            KeySigner
+	mux               sync.Mutex
+	expiresAt         time.Time
+	refreshInProgress bool
+}
+
+// KeySigner is used to sign the user's public key
+type KeySigner interface {
+	// SignSSHKey signs the user's public key and returns a signed ssh certificate
+	SignSSHKey(ctx context.Context, publicKey string) (string, error)
+}
+
+var SignSSHKeyTimeout = 30 * time.Second
+var RefreshBufferDuration = 2 * time.Hour
+
+func NewSSHKey(signer KeySigner) *SSHKey {
+	return &SSHKey{
+		signer: signer,
+	}
+}
+
+func (s *SSHKey) getSignedKey(ctx context.Context) (string, time.Time, error) {
+	log.SpanLog(ctx, log.DebugLevelInfra, "Sign cloudlet public key from Vault")
+
+	ctx, cancel := context.WithTimeout(ctx, SignSSHKeyTimeout)
+	defer cancel()
+
+	signedKey, err := s.signer.SignSSHKey(ctx, s.publicKey)
+	if err != nil {
+		return "", time.Time{}, fmt.Errorf("failed to sign cloudlet ssh key, %s", err)
+	}
+	// set expiration time
```

Comment:
shouldn't this be `// get expiration time` ?

Reply:
i'll clarify it